test/test_gpu3.cpp

```diff
@@ -8738,6 +8738,48 @@ TEST_F(NVFuserTest, AvoidCachingSliceInput) {
     }
   }
 }
+
+// Step-1 to fix https://github.com/NVIDIA/Fuser/issues/1631
+// Needs to check consumer mapped with reduction inputs in
+// isReductionOutputMapped(), otherwise compute at root domain map
+// is wrong then leads to wrong compute at position and finally
+// expr sort failed.
+// After fix, there are two persistent buffers and can be further
+// reduced to one with a following step-2 to fix the issue in resolution
+// points detection.
+TEST_F(NVFuserTest, CaRootDomainMapConsumerMappedWithReductionInput) {
+  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
+  auto fusion = fusion_ptr.get();
+  FusionGuard fg(fusion);
+
+  const int batch_size = 128;
+  const int hidden_size = 10240;
+  DataType input_dtype = DataType::Half;
+  auto tv0 = makeContigTensor(2, input_dtype);
+  auto tv1 = makeContigTensor(2, DataType::Float);
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  auto tv2 = castOp(DataType::Float, tv0);
+  auto tv3 = sum(tv2, {1});
+  auto tv4 = broadcast(tv3, {false, true});
+  auto tv5 = div(tv2, tv4);
+  auto tv6 = add(tv1, tv5);
+  auto tv7 = add(tv2, tv1);
+  fusion->addOutput(tv6);
+  fusion->addOutput(tv7);
+
+  auto options = at::TensorOptions()
+                     .dtype(data_type_to_aten(input_dtype))
+                     .device(at::kCUDA, 0);
+  auto options_fp32 = at::TensorOptions()
+                          .dtype(data_type_to_aten(DataType::Float))
+                          .device(at::kCUDA, 0);
+  auto t0 = at::randn({batch_size, hidden_size}, options);
+  auto t1 = at::randn({batch_size, hidden_size}, options_fp32);
+  FusionExecutorCache fec(std::move(fusion_ptr));
```

Comment:
Please do a more explicit validation. Here, the CA position should be validated. Or, even better is just building `ComputeAtRootDomain` and check the relevant domains are not mapped.

Also, `runFusionWithInputs` does not guarantee to reproduce the original issue. Can we just manually schedule the fusion? Or at least the normalization scheduler should be directly used.

Reply:
checked with `ComputeAtRootDomainMap`. I just manually projected tv2 in `auto tv6 = div(tv2, tv4);` to input to trigger the bug.